docs/production/deployment.md

```diff
@@ -803,6 +803,14 @@ virtualized environment (e.g. Docker, or Proxmox LXC).
 
 [supervisor-minfds]: http://supervisord.org/configuration.html?highlight=minfds#supervisord-section-values
 
+#### `nameserver`
+
+When the [S3 storage backend][s3-backend] is in use, downloads from S3 are
+proxied from nginx, which configuration requires an explicit value of a DNS
+nameserver to resolve the S3 server's hostname. Zulip defaults to using the
+resolver found in `/etc/resolv.conf`; this setting overrides any value found
+there.
```

Comment:
We should probably link to this from the S3 backend's own documentation (`uploads.md`)